lms/djangoapps/support/static/support/jsx/program_enrollments/inspector.jsx

```diff
@@ -2,59 +2,227 @@ import React from 'react';
+    <div className="ml-5">
+      <div><span className="font-weight-bold">Status</span>: {verificationStatus.status}</div>
+      {verificationStatus.error && (
+        <div>
+          <span className="font-weight-bold">Verification Error</span>: {verificationStatus.error}
+        </div>
+      )}
+      {verificationStatus.verification_expiry && (
+        <div>
+          <span className="font-weight-bold">Verification Expiration Date</span>
+          : {verificationStatus.verification_expiry}
+        </div>
+      )}
     </div>
-    <Button label="Search" type="submit" className={['btn', 'btn-primary']} />
-  </form>
+    <hr />
+  </div>
+);
+
+const renderEnrollmentsSection = enrollments => (
+  <div>
+    <h3>Program Enrollments</h3>
+    {enrollments.map(enrollment => (
+      <div key={enrollment.program_uuid} className="ml-5">
+        <h4>Program {enrollment.program_uuid} Enrollment</h4>
+        <div> <span className="font-weight-bold">Status</span>: {enrollment.status} </div>
+        <div> <span className="font-weight-bold">Created</span>: {enrollment.created} </div>
+        <div> <span className="font-weight-bold">Last updated</span>: {enrollment.modified} </div>
+        <div>
+          <span className="font-weight-bold">External User Key</span>
+          : {enrollment.external_user_key}
+        </div>
+        {enrollment.program_course_enrollments && enrollment.program_course_enrollments.map(
+          programCourseEnrollment => (
+            <div key={programCourseEnrollment.course_key} className="ml-5">
+              <h4>Course {programCourseEnrollment.course_key}</h4>
+              <div>
+                <span className="font-weight-bold">Status</span>
+                : {programCourseEnrollment.status}
+              </div>
+              <div>
+                <span className="font-weight-bold">Created</span>
+                : {programCourseEnrollment.created}
+              </div>
+              <div>
+                <span className="font-weight-bold">Last updated</span>
+                : {programCourseEnrollment.modified}
+              </div>
+              {programCourseEnrollment.course_enrollment && (
+                <div className="ml-5">
+                  <h4>Linked course enrollment</h4>
+                  <div><span className="font-weight-bold">Course ID</span>
+                    : {programCourseEnrollment.course_enrollment.course_id}
+                  </div>
+                  <div> <span className="font-weight-bold">Is Active</span>
+                    : {String(programCourseEnrollment.course_enrollment.is_active)}
+                  </div>
+                  <div> <span className="font-weight-bold">Mode / Track</span>
+                    : {programCourseEnrollment.course_enrollment.mode}
+                  </div>
+                </div>
+              )}
+            </div>
+          ))}
+      </div>
+    ))}
+    <hr />
+  </div>
+);
+
+const validateInputs = () => {
```

Comment:
@kdmccormick and @MichaelRoytman I am not happy how this function is written. But I cannot think of a better way. If you suggest we have to turn this reactjs rendering into a react component, it would take longer time than I am comfortable with. If you still think that's the way to go, can I get some help here?

Reply:
@MichaelRoytman Very detailed suggestion. I appreciate your feedback here. However, what I realized from the feedback Kyle gave, is that even when we search by edX email and username, the user also need to submit the IdP selection. The IdP selection is important to filter the social_auth_record associated with the user. That makes the two form approach having to duplicate the same dropdown.
The reason I am not happy with the `validateInput` function approach is precisely what you mentioned. If the component is not only for form submission, but also keeps a state, we should use react component `state` variable. But I am not dogmatic about this whatsoever. I think the `validateInput` I have achieves the goal of preventing users from making mistakes. 
What I want to propose is this: Let's release this tool out for feedback. The speculations about UX can be changed later. Plus, I want to move this UI into the MFE AJ is building. Let's not hold the tool up base on UX now because the number of users this tool have is limited.